core/os/time.cpp

```diff
@@ -296,14 +324,20 @@ String Time::get_datetime_string_from_datetime_dict(const Dictionary p_datetime,
 int64_t Time::get_unix_time_from_datetime_dict(const Dictionary p_datetime) const {
 	ERR_FAIL_COND_V_MSG(p_datetime.is_empty(), 0, "Invalid datetime Dictionary: Dictionary is empty");
 	EXTRACT_FROM_DICTIONARY
-	VALIDATE_YMDHMS(0)
+	VALIDATE_YMDHMS()
+	if (validation_error != "") {
+		return -1;
+	}
 	YMD_TO_DAY_NUMBER
 	return day_number * SECONDS_PER_DAY + hour * 3600 + minute * 60 + second;
 }
 
 int64_t Time::get_unix_time_from_datetime_string(String p_datetime) const {
 	PARSE_ISO8601_STRING(-1)
-	VALIDATE_YMDHMS(0)
+	VALIDATE_YMDHMS()
+	if (validation_error != "") {
+		return -1;
```

Comment:
The different return codes was an intended feature, though now that I think about it, it should've been documented. See https://github.com/godotengine/godot/pull/60215#discussion_r850865198

Reply:
So the validation error needs to return `0` there instead of `-1`. Will do.